### DISCUSSION TITLE
afl-cmin: MCDC Metric and Test Case Minimization

### DISCUSSION THREAD

**Discussion by ghost** [AFLplusplus/AFLplusplus#2118] [Q&A]
Hello,

I have a question regarding test case minimization using afl-cmin. My understanding is that afl-cmin should retain every file that takes a different path. However, in my project, when I run cmin, the value of MCDC coverage decreases and a significant number of test cases are removed.

So, my question is: Is the MCDC metric not being considered when using cmin?

**Comment by vanhauser-thc** [comment]
> My understanding is that afl-cmin should retain every file that takes a different path.

that understanding is wrong :-)
afl-cmin retains the smallest file for each reachable edge value (e.g. edge 3 with 2 visits is a unique,edge 3 with 1 visit is also unique).
there is no MCDC measurement whatsoever.
This is still the minimization from the original vanilla afl from ~2016 or so.

If you want that, then you have to implement that yourself (IMHO it would be a good thing to have, so it is something I would happily merge from a PR).
